types/id.go

```diff
@@ -1,12 +1,16 @@
 package types
 
-import sdk "github.com/cosmos/cosmos-sdk/types"
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/regen-network/regen-ledger/types/address"
+)
 
 type ModuleID struct {
 	ModuleName string
 	Path       []byte
 }
 
 func (m ModuleID) Address() sdk.AccAddress {
-	return AddressHash(m.ModuleName, m.Path)
+	return address.Module(m.ModuleName, m.Path)
```

Comment:
According to variable names, I guess that `m.Path` is a composed key. @blushi , could you confirm??
If so, we shouldn't use `address.Module` here. Instead we should implement `Composed` address function which is using length prefixes (`LengthPrefix(...)`).
@aaronc - could you confirm?

Reply:
Here, path was not a composed parameter. I refactored the code and made it more clear (I hope).